src/vcpkg-test/install.cpp

```diff
@@ -123,18 +123,43 @@ cmake_policy(POP)
     res = get_cmake_add_library_names(fmt_targets);
     CHECK(res == std::vector<std::string>{"fmt::fmt", "fmt::fmt-header-only"});
 
-    res = get_cmake_add_library_names("add_library(bar) foo_add_library(baz)");
+    res = get_cmake_add_library_names("add_library(bar) foo_add_library(baz) add_library()");
     CHECK(res == std::vector<std::string>{"bar"});
 
     res = get_cmake_add_library_names("add_library(bar) add_library(baz-bar) add_library(baz_%_bar)");
     CHECK(res == std::vector<std::string>{"bar", "baz-bar", "baz_%_bar"});
 
-    res = get_cmake_add_library_names("add_library(bar${foo})");
-    CHECK(res == std::vector<std::string>{"bar"});
-
-    res = get_cmake_add_library_names("add_library() add_library(foo) add_library(   \nbar)");
-    CHECK(res == std::vector<std::string>{"foo", "bar"});
-
     res = get_cmake_add_library_names("add_library(foo) add_library(foo) add_library(foo)");
     CHECK(res == std::vector<std::string>{"foo", "foo", "foo"});
+
+    // In the following cases, the empty list indicates the need for an explicit usage file.
+
+    auto with_var = get_cmake_add_library_names("add_library(bar${foo}) add_library(${foo}) add_library(   \nbar)");
+    CHECK(with_var.empty());
+
+    auto with_comment = get_cmake_add_library_names("add_library( # rem \n    foo)  add_library( bar# rem3)");
```

Comment:
```suggestion
    auto with_comment = get_cmake_add_library_names("add_library( # rem \n    foo)  add_library( bar# rem3\n)");
```

Reply:
or should the target name be bar# ?

Reply:
The point is to not return any target name at all from this input. It is not the generated form. If there is a comment, it likely that the heuristics won't generate useful output.

Reply:
yeah but the input should still be valid cmake shouldn't it?

Reply:
There is no benefit from perfection here. It is not a full parser.